src/index.ts

```diff
@@ -1,20 +1,28 @@
 import { h } from 'hastscript'
 import type { ShikiTransformer } from '@shikijs/types'
+import type { HTMLAttributes } from 'astro/types'
 
 export interface Options {
-  toggle?: number
+  toggle?: number;
+  button?: Omit<HTMLAttributes<'button'>, 'onclick' | 'class'> & {
+    className?: HTMLAttributes<'button'>['class'];
+  };
 }
 
 export function addCopyButton(options: Options = {}): ShikiTransformer {
   const toggleMs = options.toggle || 3000
+  const { className: optionsClassName, ...buttonAttributes } =
+    options.button || {}
+  const className = optionsClassName || 'copy'
 
   return {
     name: 'shiki-transformer-copy-button',
     pre(node) {
       const button = h(
         'button',
         {
-          class: 'copy',
+          ...buttonAttributes,
+          class: className,
```

Comment:
It would better if it used the configured class name by default,
And if no `className` is defined, fallback to `copy`

```
...button,
class: buttonAttributes?.className || 'copy'
```